src/client/theme-default/components/VPLocalSearchBox.vue

```diff
@@ -182,14 +178,18 @@ debouncedWatch(
         })
         const div = document.createElement('div')
         app.mount(div)
-        const sections = div.innerHTML.split(headingRegex)
+        const headings = div.querySelectorAll('h1, h2, h3, h4, h5, h6')
+        const promises = Array.from(headings).map(async (el) => {
+          const href = el.querySelector('a')?.getAttribute('href')
+          const anchor = href?.startsWith('#') && href.slice(1)
+          if (!anchor) return
+          let html = ''
+          while ((el = el.nextElementSibling!) && !/^h[1-6]$/i.test(el.tagName))
+            html += el.outerHTML
+          map!.set(anchor, html)
+        })
+        await Promise.all(promises)
```

Comment:
Wouldn't Promise.all be a bit redundant?

Reply:
Yeah, might be. Can we benchmark that somehow? this async thing vs headings.forEach 👀

Reply:
> Can we benchmark that somehow?

All I can think of is the difference in multiple execution times between these two methods as a benchmark test👀There is no difference in execution time after I measured it several times locally.

And what I have learned is that the `forEach` or `map` loop executes all of the iterations simultaneously.